src/include/rbtree.h

```diff
@@ -2,16 +2,162 @@
 #define RBTREE_H
 
 #include <stdint.h>
-
+/*
+The main functionality of the red-black trees is to provide a fast, 
+cheap way to map ranges of addresses to a number and acccessing by providing an address in this range.
```

Comment:
Use more detailed wording for the rationale of utilizing red-black trees for memory management. You can discuss the prior situation when red-black trees were not introduced, where sparse arrays were used, resulting in memory waste. With the introduction of red-black trees, explain the considerations and benefits below.